### PR TITLE
Implement NSCA alerting plugin

### PR DESCRIPTION
**Pull request by kristian-lesko**
Add an alerting plugin to make
it possible to dispatch tool's
runtime result as NSCA checks.

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/packages/job/freeipa-manager-docker-tox-zuul-el7/61/ : SUCCESS in 44s
- https://checklist.intgdc.com/job/compliance/job/freeipa-manager-opensource-repository-check/4/ : SUCCESS in 7s

File: ipamanager/alerting/nsca.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            return 2
+        if self.max_level >= logging.WARNING:
+            return 1
+        return 0
+
+    def _run_dispatch(self, code, message):
+        data = '%s;%s;%s;%s' % (socket.getfqdn(), self.service, code, message)
+        self.lg.debug('Dispatching NSCA data: %s', data)
+        sp = Popen((self.command, '-d', ';'),
+                   stdin=PIPE, stdout=PIPE, stderr=PIPE)
```
Comment (pbenas) [inline ipamanager/alerting/nsca.py]: have you considered implementation using a python module such as
https://pypi.org/project/pynsca/ or https://github.com/Yelp/send_nsca ?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/packages/job/freeipa-manager-docker-tox-zuul-el7/63/ : SUCCESS in 1m 04s
- https://checklist.intgdc.com/job/compliance/job/freeipa-manager-opensource-repository-check/5/ : SUCCESS in 12s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/packages/job/freeipa-manager-docker-tox-zuul-el7/64/ : SUCCESS in 1m 04s
- https://checklist.intgdc.com/job/compliance/job/freeipa-manager-opensource-repository-check/6/ : SUCCESS in 14s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/packages/job/freeipa-manager-docker-tox-zuul-el7/66/ : SUCCESS in 34s
- https://checklist.intgdc.com/job/compliance/job/freeipa-manager-merge-compliance-check/20/ : SUCCESS in 45s

**Pull request merged**